app/src/main/java/org/projectbuendia/client/ui/chart/ObsFormat.java

```diff
@@ -456,7 +456,7 @@ private boolean matches(@Nullable Object obj, String operator, String operandStr
                 case "":
                 case "=":
                 case "==":
-                    return Objects.equal(value, operand);
+                    return Objects.equal(value.uuid, operand.uuid);
```

Comment:
This doesn't look right to me.  If we let ObsValue.equals() do the comparison, it will check all its fields.  With this change, all numbers, strings, or dates will compare equal even though they shouldn't.

What was the root cause of the incorrect behaviour you were seeing, and how did this change fix it?